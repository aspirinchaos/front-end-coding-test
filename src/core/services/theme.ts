import { DefaultTheme, IBreakpoint } from 'styled-components';

import { min, max } from './styled-responsive';

const BREAKPOINTS: IBreakpoint = {
  xs: 576,
  sm: 600,
  md: 992,
  lg: 1200,
  xl: 1700,
};

const theme: DefaultTheme = {
  breakpoints: BREAKPOINTS,
  colors: {
    main: '#e85a4f',
    secondary: '#8e8d8a',
    body: '#eae7dc',
  },
  font: 'Poppins,sans-serif',
  min,
  max,
};

export default theme;
