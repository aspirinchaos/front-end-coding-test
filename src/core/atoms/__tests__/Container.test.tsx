import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import theme from 'core/services/theme';

import { Container } from '..';

describe('Container atom', () => {
  it('renders correctly', () => {
    const { container } = render(<Container theme={theme}>Content</Container>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('has correct styles', () => {
    const { container } = render(<Container theme={theme}>Content</Container>);
    expect(container.firstChild).toHaveStyleRule('width', '100%');
    expect(container.firstChild).toHaveStyleRule('padding', '0 50px', {
      media: `(min-width:${theme.breakpoints.xl}px)`,
    });
  });
});
