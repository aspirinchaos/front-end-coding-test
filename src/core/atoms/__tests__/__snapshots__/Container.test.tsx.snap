// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container atom renders correctly 1`] = `
.c0 {
  width: 100%;
  max-width: 1200px;
  margin: auto;
}

@media (min-width:1700px) {
  .c0 {
    max-width: -webkit-fit-content;
    max-width: -moz-fit-content;
    max-width: fit-content;
    padding: 0 50px;
  }
}

<section
  class="c0"
>
  Content
</section>
`;
